lvmd/vgservice.go

```diff
@@ -111,14 +163,53 @@ func (s *vgService) send(server proto.VGService_WatchServer) error {
 		if err != nil {
 			return err
 		}
-		if dc.Default {
-			res.FreeBytes = vgFree
+
+		switch dc.Type {
+		case TypeThick:
+			if dc.Default {
+				res.FreeBytes = vgFree
+			}
+			res.Items = append(res.Items, &proto.WatchItem{
+				DeviceClass: dc.Name,
+				FreeBytes:   vgFree,
+				SizeBytes:   vgSize,
+			})
+		case TypeThin:
+			tpi := &proto.ThinPoolItem{}
+			pool, err := vg.FindPool(dc.ThinPoolConfig.Name)
+			if err != nil {
+				return status.Error(codes.Internal, err.Error())
+			}
+			tpu, err := pool.Free()
+			if err != nil {
+				return status.Error(codes.Internal, err.Error())
+			}
+			opb := uint64(math.Floor(dc.ThinPoolConfig.OverprovisionRatio*float64(tpu.SizeBytes))) - tpu.VirtualBytes
+			if dc.Default {
+				// TODO (leeavg): remove this after removal of support for inline ephemeral volumes
+				res.FreeBytes = opb
+			}
+
+			// used for updating prometheus metrics
+			tpi.DataPercent = tpu.DataPercent
+			tpi.MetadataPercent = tpu.MetadataPercent
+
+			// used for annotating the node for capacity aware scheduling
+			tpi.OverprovisionBytes = opb
+
+			// TODO (leelavg): one can calculate freebytes by using data, metdata and size values,
```

Comment:
@leelavg 
There are some `TODO` in your code. Would you please check those comments and remove them if aren't needed.

Reply:
done.